### PR TITLE
When safety_check_fail, also embed hint msg in the abort function name

### PR DESCRIPTION
**Pull request by interwq**
because there are cases only logging crash stack traces.

### CODE REVIEW

File: src/safety_check.c [hunk 1]
```diff
@@ -20,6 +20,20 @@ void safety_check_set_abort(safety_check_abort_hook_t abort_fn) {
 	safety_check_abort = abort_fn;
 }
 
+/*
+ * In addition to malloc_write, also embed hint msg in the abort function name
+ * because there are cases only logging crash stack traces.
+ */
+static void
+safety_check_detected_heap_corruption___run_address_sanitizer_build_to_debug(const char *buf) {
```
Comment (interwq) [inline src/safety_check.c:28]: @davidtgoldblatt @guangli-dai @notdevil open to suggestions here

**Comment by interwq** [comment]
(The CI failure will be fixed by #2581)

**Comment by davidtgoldblatt** [comment]
A++ function naming, would review again.

**Review by guangli-dai — APPROVED** [review]
Great naming!

**Pull request merged**